gmso/formats/lammpsdata.py

```diff
@@ -654,13 +719,437 @@ def _get_ff_information(filename, unit_style, topology):
+        out_file.write(
+            "{}\t{:7.5f}\t{:7.5f}\t#{}\t{}\t{}\n".format(
+                idx + 1,
+                _parameter_converted_to_float(
+                    angle_type.parameters["k"], base_unyts, cfactorsDict
+                ),
+                angle_type.parameters["theta_eq"]
+                .to(u.degree)
+                .value,  # write equilibrium values in degrees
+                *angle_type.member_types,
+            )
+        )
+
+
+from gmso.core.views import get_sorted_names
+
+
+def _write_dihedraltypes(out_file, top, base_unyts, cfactorsDict):
+    """Write out dihedrals to LAMMPS file."""
+    test_dihtype = top.dihedrals[0].dihedral_type
+    out_file.write(f"\nDihedral Coeffs #{test_dihtype.name}\n")
+    param_labels = map(
+        lambda x: f"{x} ({test_dihtype.parameters[x].units})",
+        test_dihtype.parameters,
+    )
+    out_file.write("#\t" + "\t".join(param_labels) + "\n")
+    indexList = list(top.dihedral_types(filter_by=pfilter))
+    index_membersList = [
+        (dihedral_type, get_sorted_names(dihedral_type))
+        for dihedral_type in indexList
+    ]
+    index_membersList.sort(key=lambda x: ([x[1][i] for i in [1, 2, 0, 3]]))
+    for idx, (dihedral_type, members) in enumerate(index_membersList):
+        out_file.write(
+            "{}\t{:8.5f}\t{:8.5f}\t{:8.5f}\t{:8.5f}\t# {}\t{}\t{}\t{}\n".format(
+                idx + 1,
+                *[
+                    _parameter_converted_to_float(
+                        dihedral_type.parameters[parameterStr],
+                        base_unyts,
+                        cfactorsDict,
+                    )
+                    for parameterStr in ["k1", "k2", "k3", "k4"]
+                ],
+                *members,
+            )
+        )
+
+
+def _write_impropertypes(out_file, top, base_unyts, cfactorsDict):
+    """Write out impropers to LAMMPS file."""
+    # TODO: Use any accepted lammps parameters
+    test_imptype = top.impropers[0].improper_type
+    out_file.write(f"\nImproper Coeffs #{test_imptype.name}\n")
+    param_labels = map(
+        lambda x: f"{x} ({test_imptype.parameters[x].units})",
+        test_imptype.parameters,
+    )
+    out_file.write("#\t" + "\t".join(param_labels) + "\n")
+    for idx, improper_type in enumerate(top.improper_types(filter_by=pfilter)):
+        out_file.write(
+            "{}\t{:7.5f}\t{:7.5f}\n".format(
+                idx + 1,
+                _parameter_converted_to_float(
+                    improper_type.parameters["k"], base_unyts, cfactorsDict
+                ),
+                improper_type.parameters["chieq"]
+                .to(u.degree)
+                .value,  # write to degrees
+                *improper_type.members,
+            )
+        )
+
+
+def _write_site_data(out_file, top, atom_style, base_unyts, cfactorsDict):
+    """Write atomic positions and charges to LAMMPS file.."""
+    out_file.write("\nAtoms\n\n")
+    if atom_style == "atomic":
```

Comment:
what about other styles (e.g. DPD)

Reply:
Will be addressed in later PR's on a case by case basis.